plugins/inputs/jti_openconfig_telemetry/jti_openconfig_telemetry.go

```diff
@@ -279,13 +291,18 @@ func (m *OpenConfigTelemetry) collectData(
 					// Print final data collection
 					m.Log.Debugf("Available collection for %s is: %v", grpcServer, dgroups)
 
-					tnow := time.Now()
+					timestamp := time.Now()
 					// Iterate through data groups and add them
 					for _, group := range dgroups {
+						if m.TimestampSource == "data" {
+							// OpenConfig timestamp is in milliseconds since epoch
+							timestamp = time.UnixMilli(int64(group.data["_timestamp"].(uint64)))
```

Comment:
I would really like to check the assertion here to prevent a panic
```suggestion
							ts, ok := group.data["_timestamp"].(uint64)
							if ok {
								timestamp = time.UnixMilli(int64(ts))
							} else {
								m.Log.Warnf("invalid type %T for _timestamp %v", group.data["_timestamp"], group.data["_timestamp"])
							}
```

As background, my experience for GNMI tells me that behavior among vendors (or even between models) is not necessarily conforming to any expectations and might vary. So better be safe than sorry... Maybe the warning could be a "warn only once" thing...

Reply:
agreed on type conversion - done. 

Given this is opt-in, let's log each time and see if we get any issues that come through.